### PR TITLE
Add nat gateway setup

### PR DESCRIPTION
**Pull request by lpalovsky**
This PR adds NAT Gateway setup into workload zone deployment process, which is 
required for SUT to be able to reach scc repos. 

- Related ticket: https://jira.suse.com/browse/TEAM-9311
- Verification run: https://openqaworker15.qa.suse.cz/tests/285787#step/sdaf_deploy_hanasr/38

* VR fails but passes playbook `playbook_01_os_base_config.yaml` which previously failed due to SCC connection issues.

### CODE REVIEW

File: lib/sles4sap/sdaf_deployment_library.pm [hunk 1]
```diff
@@ -414,6 +415,9 @@ sub set_common_sdaf_os_env {
     $args{sdaf_tfstate_storage_account} //= get_required_var('SDAF_TFSTATE_STORAGE_ACCOUNT');
     $args{sdaf_key_vault} //= get_required_var('SDAF_KEY_VAULT');
 
+    # This is used later filling up tfvars files.
```
Comment (lpalovsky) [inline lib/sles4sap/sdaf_deployment_library.pm:418]: Only a temporary solution, tfvars templates will be reworked once the whole base deployment works

File: lib/sles4sap/sdaf_deployment_library.pm [hunk 2]
```diff
@@ -414,6 +415,9 @@ sub set_common_sdaf_os_env {
...
+    set_var('SDAR_REGION_CODE', $args{sdaf_region_code});
```
Comment (mpagot) [inline lib/sles4sap/sdaf_deployment_library.pm:419]: there was no any `tfvars` before that needs to know about region? Is there any other setting about region?

Comment (lpalovsky) [reply lib/sles4sap/sdaf_deployment_library.pm:419]: It's in the public IP and NAT GW resource name :

https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/19461/files#diff-f72c92100b8dddfc10ceb79d42a1fb58d2bf3de5d03c946fbcf85e51506526fbR96

But maybe I could try to remove the whole naming line. With other resources the naming is done internally.

Comment (lpalovsky) [reply lib/sles4sap/sdaf_deployment_library.pm]: So on last MS meeting they mentioned that naming is mandatory. :/

**Review by mpagot — APPROVED** [review]
LGTM

**Review by BillAnastasiadis — APPROVED** [review]
LGTM

**Review by alvarocarvajald — APPROVED** [review]
LGTM

**Pull request merged**